test/test_integration_systemd.rb

```diff
@@ -59,4 +56,28 @@ def test_systemd_watchdog
     stop_server
     assert_match(socket_message, "STOPPING=1")
   end
+
+  private
+
+  def restart(signal, workers: 2)
```

Comment:
Stylistically, I think every test should end with `assert` or, rarely, a cleanup step or two. This method is basically a set of assertions, so I think it would be appropriate to call it `assert_restarts_with_systemd` or similar.